From b9df2b5a41782c3779bf3604c00bf00515e7888e Mon Sep 17 00:00:00 2001
From: Li Jun <B47624@freescale.com>
Date: Fri, 7 Mar 2014 10:15:27 +0800
Subject: [PATCH 0753/1074] ENGR00302223 usb: charger: increase wait time
 before check CHRG_DETECTED bit

One customer reported an issue that sometimes SW cannot get right charger
detection status on their HW, that is: CHRG_DETECTED bit sometimes is still
0 after 40ms wait time, increase the wait time to be 100ms can resolve the
issue with customer's HW and usb charger. Per usb PHY IC owner's comments,
the required wait time depends on charger's cap of the charger, bigger cap
need more wait time, BC spec only define the min wait time 40mSx, not define
max wait time, so it's ok to have 100ms wait time.

Acked-by: Peter Chen <peter.chen@freescale.com>
Signed-off-by: Li Jun <b47624@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 drivers/power/imx6_usb_charger.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/power/imx6_usb_charger.c b/drivers/power/imx6_usb_charger.c
index d94a302..81a8064 100644
--- a/drivers/power/imx6_usb_charger.c
+++ b/drivers/power/imx6_usb_charger.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2013 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2013-2014 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * The code contained herein is licensed under the GNU General Public
  * License. You may obtain a copy of the GNU General Public License
@@ -142,7 +142,7 @@ static int imx6_usb_charger_detect(struct usb_charger *charger)
 	regmap_write(regmap, HW_ANADIG_USB1_CHRG_DETECT_CLR,
 			BM_ANADIG_USB1_CHRG_DETECT_CHK_CONTACT |
 			BM_ANADIG_USB1_CHRG_DETECT_CHK_CHRG_B);
-	msleep(45);
+	msleep(100);
 
 	/* Check if it is a charger */
 	regmap_read(regmap, HW_ANADIG_USB1_CHRG_DET_STAT, &val);
-- 
1.7.5.4

